File: backend/app/models/loan.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime

class LoanBase(BaseModel):
    title: str
    amount: float
    interest_rate: float
    term_months: int
    start_date: datetime
    description: Optional[str] = None

class LoanCreate(LoanBase):
    pass

class Loan(LoanBase):
    id: str
    user_id: str
    created_at: datetime
    total_amount: float  # Principal + Interest
    monthly_payment: float
    status: str = "active"  # active, paid, defaulted

    class Config:
        orm_mode = True

File: frontend/src/types/loan.ts
export interface Loan {
  id: string;
  user_id: string;
  title: string;
  amount: number;
  interest_rate: number;
  term_months: number;
  start_date: string;
  description?: string;
  created_at: string;
  total_amount: number;
  monthly_payment: number;
  status: 'active' | 'paid' | 'defaulted';
}

export interface LoanFormData {
  title: string;
  amount: number;
  interest_rate: number;
  term_months: number;
  start_date: string;
  description?: string;
}